packages/styled-components/src/models/StyledComponent.js

```diff
@@ -173,7 +174,10 @@ function useStyledComponentImpl<Config: {}, Instance>(
     propsForElement.style = { ...attrs.style, ...props.style };
   }
 
-  propsForElement.className = Array.prototype
+  propsForElement[
+    // handle custom elements which React doesn't properly alias
+    isTargetTag && domElements.indexOf(elementToBeCreated) === -1 ? 'class' : 'className'
```

Comment:
This seems... rather odd. I wasn't aware that `class` was allowed in principal at all actually. Couldn't we always use `class` for `isTargetTag` then?

One nit here btw: Accessing the list of element names by `indexOf` is _relatively_ expensive.

Reply:
I think if you pass `class` to a recognized tag React will warn :/